lib/apiservers/engine/backends/backends.go

```diff
@@ -359,3 +337,59 @@ func loadRegistryCACerts() {
 func EventService() *events.Events {
 	return eventService
 }
+
+func (d *dynConfig) RegistryCheck(u *url.URL) (wl bool, bl bool, insecure bool) {
+	d.Lock()
+	defer d.Unlock()
+
+	// update config
+	if err := d.update(); err != nil {
+		log.Warnf("error updating config: %s", err)
+	}
+
+	us := u.String()
+	wl = len(d.Whitelist) == 0 || d.Whitelist.Match(us)
+	bl = len(d.Blacklist) == 0 || !d.Blacklist.Match(us)
+	insecure = d.Insecure.Match(us)
+	return
+}
+
+func (d *dynConfig) Update() error {
+	d.Lock()
+	defer d.Unlock()
+
+	return d.update()
+}
+
+func (d *dynConfig) update() error {
+	// update config
+	c, err := d.src.Get()
+	if err != nil {
+		return err
+	}
+
+	if c == nil {
+		return nil
+	}
+
+	newcfg, err := d.merger.Merge(vchConfig.Cfg, c)
```

Comment:
We don't have to, but we would have to create a copy of the config every time we do a merge then.